### PR TITLE
Migrate lint and format to Makefile

### PR DESCRIPTION
**Pull request by kwridan**
### Short description 📝

- To ease contributions to those that don't have a ruby setup, the SwiftLint and Swiftformat shell calls have been migrated to the Makefile
- The SwiftFormat config was updated to exclude one of the fixtures that is added as a submodule
- The github token was moved to the github step to avoid requiring it for all makefile incovations

### How to test the changes locally 🧐

- Modify any Swift source file to introduce formatting errors
- Run `make lint`
- Verify errors are flagged and the exit code is non-zero
- Run `make lint-fix`
- Verify the formatting errors are corrected
- Run `make lint`
- Verify it passes and the exist code is zero

### Contributor checklist ✅

- [ ] The code has been linted using run `./fourier lint tuist --fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [ ] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Review by danyf90 — APPROVED** [review]

**Review by danyf90 — APPROVED** [review]

**Review by danyf90 — CHANGES_REQUESTED** [review]
Shouldn't we also replace it in `tuist/tuist/.github/workflows/tuist.yml`?
How hard would it be to get rid of the other fourier lint commands as well?

**Comment by kwridan** [comment]
> Shouldn't we also replace it in tuist/tuist/.github/workflows/tuist.yml

Good point would need to have 2 separate actions `lint` `lint-fix` or something like that one that works in lint mode one that applies fixes.

> How hard would it be to get rid of the other fourier lint commands as well?

Fourier lint also lints ruby code - hence why originally I made it `tuist_lint` 😅 - not too sure on what's needed for the ruby side of things so kept that as is.

**Review by danyf90 — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -20,7 +19,14 @@ generate:
 run:
 	swift build
 	.build/debug/tuist $(ARGS)
+lint-fix:
```
Comment (pepicrft) [inline Makefile:22]: @kwridan I'll merge [this one](https://github.com/tuist/tuist/pull/5451) first and then push a commit to this one extracting the logic into their own scripts.

File: Makefile [hunk 2]
```diff
@@ -10,3 +10,7 @@ run:
 	./make/tasks/run.sh $(ARGS)
 generate/cloud-openapi-code:
 	./make/tasks/generate/cloud-openapi-code.sh
+lint-fix:
+	./make/tasks/lint-fix.sh
+lint:
+	./make/tasks/lint.sh
```
Comment (danyf90) [inline Makefile:16]: I would make `lint.sh` take a --fix parameter and handle the change there

Not sure if it's also possible to define arguments for make (I guess not)

**Review by danyf90 — APPROVED** [review]

**Pull request merged**